### ISSUE TITLE
Patient id not populated in reference app

### ISSUE THREAD

**Issue by yrimal** [medic/cht-core#6383]
**Describe the bug**
When I install ANC Reference App on CHT-Core and submit Pregnancy registration form, `patient_id` field is not populated. 

**To Reproduce:**
1. On your development/deployment server, Install ANC reference app (cht-core/config/default) using medic-conf. 
2. Create required persona and one chw who can submit pregnancy registration form.
3. Login with CHW created in step 2 and submit one pregnancy form. 
4. Pick GUID of the form submitted in Step 3.
5. Go to couchdb and browse the GUID from step 4. 
6. Search for `patient_id` field. All 3 occurance of `patient_id` is empty. 

This field is used for analysis once migrated to postgres and is expected to populate. Also, it is easier to use than other options.

**Expected behavior**
`patient_id` field should be populated in document with correct patient id for whom this form was submitted. 

**Environment**
- Instance: Any instances with ANC Reference app
- Browser: Chrome
- Client platform: WSL
- App: Reference App
- Version: 3.8

**Additional context**

**Comment by dianabarsan** [comment]
@yrimal Is the patient you are submitting the form for created on the device (the phone) right before submitting the pregnancy registration form? 
If not, does the patient you are submitting the form for have a `patient_id` field?

**Comment by yrimal** [comment]
@dianabarsan , Yes, it was created right before submitting the pregnancy registration form. Both done from webapp but after Manually syncing the contact.

**Comment by dianabarsan** [comment]
@yrimal In order for your contact on the device to have a `patient_id` you need to sync twice. Once to upload and once to download the updated document. 
To have a patient_id on the report, you should create the contact, sync to upload, wait for the contact to be updated by Sentinel (maybe check in the DB or logged in as an admin) and received a `patient_id`, sync and then create your report.

**Comment by yrimal** [comment]
I tested it the way you mentioned and that sounds right @dianabarsan .

However, it is a likely situation that when registering new pregnancy, CHW first register person and then submit pregnancy registration form after that without update happening. Since we don't force user to update before submitting form, there'll be data integrity issue.

Will it be possible for sentinel or any other module in application to update patient_id field in next run?

**Comment by dianabarsan** [comment]
The you should use `fields.patient_uuid` in analytics instead of or in conjunction with `patient_id`. It's impossible to have a guarantee that you will always have online fields populated in offline workflows.

**Comment by 1yuv** [comment]
Hi @dianabarsan , we saw a similar case like this in one of our prod instance. The patient id is now assigned to this patient, but scheduled messages are still missing reference to the pateint id. Detail [here](https://github.com/medic/config-moh-nepal/issues/1166#issuecomment-1934676969).

**Comment by dianabarsan** [comment]
Nice catch @1yuv . You are right and I have one suspicion. 
I'm checking the same message in the admin app, and in the admin app it `patient_id` does get populated in SMS. My only guess was that in Webapp we're doing something different, and upon checking it seems that we're only passing formatted fields to the message context, and these formatted fields only include visible fields. In your case, patient_id is a hidden field. Can you try deleting `patient_id` from the `hidden_fields` property in the doc and reloading the page to see if this fixes it?

**Comment by dianabarsan** [comment]
I think this is just a visual problem in Webapp, that when the SMS is sent the correct data will be attached.

**Comment by 1yuv** [comment]
I suspect and wish it should be the case @dianabarsan since we use translation key to store message, otherwise it'd have stored a message without patient_id. At the time of sending message, probably gateway generates the correct translation and it should have patient_id attached by then. However, I beleive a report also loads translation key everytime you load report tabm in that case, shouldn't it show the correct message? Is this caching issue in Reports tab?

**Comment by dianabarsan** [comment]
I looked more into this, and it turns out it;s a bug in Webapp in how we populate the patient when viewing data records. This is valid across any config.

**Comment by dianabarsan** [comment]
I looked more into it, and it seems like it's indeed a bug in how we generate messages in sentinel as well. 
so far, only the admin app page generates messages correctly. 

I'm adding this to 4.7 to get fixed.

**Comment by dianabarsan** [comment]
Merged to `master`. 
@garethbowen I know we discussed about backporting this to 4.6 / 4.5 to get this faster to our prod instances. Just wanting to confirm that this is still happening.

**Comment by garethbowen** [comment]
I'm happy to do the backport if it'll get used.

@1yuv If you had the option of installing 4.7.0 (out soon!) or 4.6.1, both of which had this fix, which would you choose?
